src/Compilers/CSharp/Portable/Symbols/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -82,6 +88,37 @@ protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymb
 
                 return parameter;
             }
+
+            static bool checkUnderspecifiedGenericExtension(TypeSymbol underlyingType, ImmutableArray<TypeParameterSymbol> typeParameters,
+                BindingDiagnosticBag diagnostics, Location location)
+            {
+                var usedTypeParameters = PooledHashSet<TypeParameterSymbol>.GetInstance();
+                underlyingType.VisitType(collectTypeParameters, arg: usedTypeParameters);
+
+                bool anyUnusedTypeParameter = false;
```

Comment:
>anyUnusedTypeParameter [](http://example.com/codeflow?start=21&length=22)

Consider removing this local. Return value of this local function is never used, therefore, there is no need to collect and return this information